judge/contest_format/ioi.py

```diff
@@ -0,0 +1,100 @@
+from datetime import timedelta
+
+from django.core.exceptions import ValidationError
+from django.db import connection
+from django.template.defaultfilters import floatformat
+from django.urls import reverse
+from django.utils.html import format_html
+from django.utils.safestring import mark_safe
+from django.utils.translation import gettext_lazy
+
+from judge.contest_format.default import DefaultContestFormat
+from judge.contest_format.registry import register_contest_format
+from judge.timezone import from_database_time
+from judge.utils.timedelta import nice_repr
+
+
+@register_contest_format('ioi')
+class IOIContestFormat(DefaultContestFormat):
+    name = gettext_lazy('IOI')
+    config_keys = ('no_cumtime',)
+    '''
+        no_cumtime: Specify True if time penalities are to be ignored. Defaults to False.
+    '''
+
+    @classmethod
+    def validate(cls, config):
+        if config is None:
+            return
+
+        if config is not isinstance(config, dict):
+            raise ValidationError('ioi-styled contest expects no config or dict as config')
+
+        for key in config.keys():
+            if key not in cls.config_keys:
+                raise ValidationError('unknown config key "%s"' % key)
+
+    def __init__(self, contest, config):
+        super().__init__(contest, config)
+        # We will be treating this like a dictionary, so ensure it is not None
+        self.config = self.config or {}
+        for key in self.config_keys:
+            if key not in self.config.keys():
+                self.config[key] = False
```

Comment:
Maybe have default values in `self.config_keys`. In fact, just be like `self.config = self.config_defaults.copy(); self.config.update(config)`.